tasks/main.yml

```diff
@@ -19,52 +19,68 @@
     name: "{{ __vpn_packages }}"
     state: present
 
-- name: Enforce default encryption type as needed
-  set_fact:
-    vpn_connections: |
-      {% for tunnel in vpn_connections | d([]) %}
-      {%   set _ = tunnel.__setitem__(
-        "enc_type", tunnel.enc_type | d(vpn_enc_type)
-        ) %}
-      {% endfor %}
-      {{ vpn_connections | d([]) }}
+- name: Controller tasks
+  block:
+    - name: Enforce default encryption type as needed
+      set_fact:
+        vpn_connections: |
+          {% for tunnel in vpn_connections | d([]) %}
+          {%   set _ = tunnel.__setitem__(
+            "enc_type", tunnel.enc_type | d(vpn_enc_type)
+            ) %}
+          {% endfor %}
+          {{ vpn_connections | d([]) }}
 
-- name: copy psk values from a file
-  set_fact:
-    vpn_connections: |
-      {% for tunnel in vpn_connections | d([]) %}
-      {%   if tunnel.enc_type == "psk" %}
-      {%     if "shared_key_content" not in tunnel %}
-      {%       if "shared_key_src" in tunnel %}
-      {%         set _ = tunnel.__setitem__(
-        "shared_key_content", lookup('file', tunnel.shared_key_src)
-        ) %}
-      {%       endif %}
-      {%     endif %}
-      {%   endif %}
-      {% endfor %}
-      {{ vpn_connections | d([]) }}
+    - name: Ensure cert_names are populated when enc_type is cert
+      vars:
+        failure: >-
+          {%- for tunnel in vpn_connections | d([]) %}
+          {%   if tunnel.enc_type == 'cert' and tunnel.hosts.items()%}
+          {%     for key, value in tunnel.hosts.items() | d({})%}
```

Comment:
I think this default will not work - instead, you should ensure in a prior `if` clause that `tunnel.hosts` has elements